packages/test-sync/src/commonTest/kotlin/io/realm/kotlin/test/mongodb/common/PBSProgressListenerTests.kt

```diff
@@ -293,33 +275,25 @@ class ProgressListenerTests {
     private suspend fun Realm.writeSampleData(count: Int, idOffset: Int = 0, timeout: Duration? = null) {
         write {
             for (i in idOffset until count + idOffset) {
-                copyToRealm(SyncObjectWithAllTypes().apply { stringField = "Object $i" })
+                copyToRealm(SyncObjectWithAllTypes().apply { binaryField = Random.nextBytes(1_000_000) })
             }
         }
         timeout?.let {
             assertTrue { syncSession.uploadAllLocalChanges(timeout) }
         }
     }
 
-    // Operator that will return a flow that emits an incresing integer on each completion event
+    // Operator that will return a flow that emits an increasing integer on each completion event
     private fun Flow<Progress>.completionCounter(): Flow<Int> =
-        filter { it.isTransferComplete }
-            .distinctUntilChanged()
-            // Increment completed count if we are done transferring and the amount of bytes has
-            // increased
-            .scan(0UL to 0) { (bytes, completed), progress ->
-                if (progress.isTransferComplete && progress.transferableBytes > bytes) {
-                    (progress.transferredBytes to completed + 1)
-                } else {
-                    (bytes to completed)
-                }
-            }
-            .drop(1)
-            .map { (_, completed) -> completed }
+        map {
```

Comment:
This flow not does something something completely different. Now it just counts the number of events!?